source/guides/first-steps/task-script.md

```diff
@@ -1,4 +1,121 @@
-Your First Task Script (NOT YET WRITTEN)
+Your First Task Script
 ----------------------
 
-**TODO: Write-up that guides the user through writing a basic task script (just a task with a narrate line) using VS-Code, emphasizing things like file extension and placement. Also run it in-game with `/ex run`.**
+```eval_rst
+.. contents:: Table of Contents
+    :local:
+```
+
+#### Task script basics
+
+A task script is a stand-alone script that can be run either via in-game command with the `/ex` command, as explained in [the guide to the /ex command](https://guide.denizenscript.com/guides/first-steps/ex-command.html), or by the `run` command in a script.
+
+Task scripts will run all of the Denizen commands that they include. A task script can be simple, it can be complicated, and it can even run other task scripts. This is useful to create logic-based script chains, like dialogue options or random content.
+
+#### Task script syntax
+
+Here's an example of a basic task script.
+
+```dscript_green
+example_task:
+    type: task
+    script:
+    - narrate "This is a basic task script!"
+```
+
+This script will narrate the text:
+
+```
+This is a basic task script!
+```
+
+to the player attached to the script. If you use the `/ex` command to run this script - specifically, by running `/ex run example_task` in-game, *you* will be the attached player.
+
+### Building your first task script in VS Code
+
+Previously, you've learned how to create an environment in VS Code, the editor that we recommend for writing Denizen scripts.
```

Comment:
The text `how to create an environment in VS Code` should back-link to the relevant page, just in case a reader skipped ahead and might want to backtrack momentarily here.